internal/autodetect/main.go

```diff
@@ -29,14 +34,17 @@ var _ AutoDetect = (*autoDetect)(nil)
 type AutoDetect interface {
 	OpenShiftRoutesAvailability() (openshift.RoutesAvailability, error)
 	PrometheusCRsAvailability() (prometheus.Availability, error)
+	RBACPermissions() (autoRbac.Availability, error)
 }
 
 type autoDetect struct {
-	dcl discovery.DiscoveryInterface
+	ctx      context.Context
+	dcl      discovery.DiscoveryInterface
+	reviewer *rbac.Reviewer
 }
 
 // New creates a new auto-detection worker, using the given client when talking to the current cluster.
-func New(restConfig *rest.Config) (AutoDetect, error) {
+func New(restConfig *rest.Config, ctx context.Context, reviewer *rbac.Reviewer) (AutoDetect, error) {
```

Comment:
suggestion: it's never good to keep context in a struct, we should instead pass context when its needed